services/provider/consumerstatus/status.go

```diff
@@ -0,0 +1,41 @@
+package consumerstatus
+
+import (
+	pb "github.com/red-hat-storage/ocs-operator/v4/services/provider/pb"
+)
+
+type ConsumerStatus interface {
+	GetClusterVersion() string
+	GetOperatorVersion() string
+
+	SetClusterVersion(string) ConsumerStatus
+	SetOperatorVersion(string) ConsumerStatus
+}
+
+type StatusInfo struct {
+	*pb.ReportStatusRequest
+}
+
+var _ ConsumerStatus = &StatusInfo{}
```

Comment:
Further down the review, I can see we need to assign an existing Request structure to the wrapper  so maybe keeping this as a pointer is not a bad approach (to prevent a copy operation)

Reply:
fails to compile if the struct doesn't satisfy the interface https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance